### PR TITLE
Adding support for updating encryption flag to AccountUpdateTool

### PR DESCRIPTION
**Pull request by nsivabalan**
Note: Yet to add tests.

Coding style applied
SLA: 5 mins

### CODE REVIEW

File: ambry-tools/src/main/java/com.github.ambry/account/AccountUpdateTool.java [hunk 1]
```diff
@@ -116,10 +124,23 @@
   public static void main(String args[]) throws Exception {
     OptionParser parser = new OptionParser();
 
+    ArgumentAcceptingOptionSpec<String> typeOfOperationOpt =
+        parser.accepts("typeOfOperation", "Type of operation. Supported operations: UpdateEntireJson, EnableEncryption")
+            .withRequiredArg()
+            .describedAs("typeOfOperation")
+            .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> enableEncryptionForOpt = parser.accepts("updateEncryptionFor",
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/account/AccountUpdateTool.java]: not sure why we need a different option for this - if you update the account json in the file path correctly, should be enough? This is what the tool seems to do for any other change so I'm not sure why encryption should be treated any differently. Also introduces unnecessary complexity in terms of arguments/input etc. Any problems doing it that way?

Comment (nsivabalan) [reply ambry-tools/src/main/java/com.github.ambry/account/AccountUpdateTool.java]: I want to simplify the account update tool going forward. Requiring to send in the entire json as input is not trivial and error prone. We have to fetch the account json from helix, edit it manually and then send it as input to the tool. But from now on, we might have lot of use-cases where in we need to update some specific properties of an account or container like encryption field, media scan enabled, cache related settings and so on. Just hold on for some time. I will try to make this argument generic which could be extended further by just introducing new type of operation. For eg: UpdateMediaScan, and so on. The argument "enableEncryptionForOpt" (will have to be renamed) will take in accountContainerPropertyValuePair( in this context, account1:container1:true,account2:container1:false). Depending on the operation type, the value will be interpreted. In this case, it refers to media scan. Will let you know once the patch is updated.

Comment (vgkholla) [reply ambry-tools/src/main/java/com.github.ambry/account/AccountUpdateTool.java]: Sounds good - however before you make any changes, lets have a chat with @cgtz.

**Comment by cgtz** [comment]
I was thinking that, instead of adding extra code to support every single container option, could we find a way to fetch an account by name, open the existing json in a text editor (i.e. the one defined by the $EDITOR env variable) and let the user edit whatever fields they desire? I think this would also let us do stuff like adding containers easily.

**Comment by vgkholla** [comment]
@cgtz that sounds great ! What's the effort though?

If that is too much, I think we should just ensure that the tool uploads all the fields correctly (i.e. no reason to treat encryption as special). If it already does that correctly, I propose that we close this pull request. (cc @nsivabalan )

**Comment by vgkholla** [comment]
@nsivabalan Please see comments above. If this pull request is no longer required, let's close it

**Comment by nsivabalan** [comment]
Will see the efforts required to get what casey suggesed. If its going to take time, I would still go with the tool support for updating encryption, cacheability values. Downloading the json, editing manually and re-uploading is non-trivial and error prone in my opinion

**Comment by vgkholla** [comment]
@nsivabalan if we have to update the tool everytime we add a field, then thats not sustainable either. I don't think these fields (encryption, cacheability) are any different from any of the other fields. If we can't do what @cgtz suggested, we should drop this and come with something thats more sustainable and maintainable.

**Pull request closed without merging**